File: aula_03/python/Bhaskara.py
import math

class Bhaskara:
    def __init__(self):
        self.__a = 0.0
        self.__b = 0.0
        self.__c = 0.0
        self.__delta = 0.0

    def calcular(self, a, b, c):
        self.__a = a
        self.__b = b
        self.__c = c

        self.__calcularDelta()
        self.__validarDelta()
        x1 = self.__calcularX1()
        x2 = self.__calcularX2()

        if(x1 == x2):
            return [x1]
        else:
            return [x1, x2]

    def __calcularDelta(self):
        self.__delta = (self.__b * self.__b) - (4 * self.__a * self.__c)

    def __validarDelta(self):
        if(self.__delta < 0):
            raise Exception("ERRO: Valores de A, B e C resultam num Delta negativo.\n")

    def __calcularX1(self):
        return (
            ((self.__b * -1) + math.sqrt(self.__delta))
            /
            (2 * self.__a)
        )

    def __calcularX2(self):
        return (
            ((self.__b * -1) - math.sqrt(self.__delta))
            /
            (2 * self.__a)
        )

File: aula_03/java/Bhaskara.java
public class Bhaskara
{
    private double a;
    private double b;
    private double c;
    private double delta;

    public double[] calcular(double a, double b, double c) throws Exception
    {
        this.a = a;
        this.b = b;
        this.c = c;

        this.calcularDelta();
        this.validarDelta();
        double x1 = this.calcularX1();
        double x2 = this.calcularX2();

        if(x1 == x2) {
            double[] resultado = {x1};
            return resultado;
        } else {
            double[] resultado = {x1, x2};
            return resultado;
        }
    }

    private void calcularDelta()
    {
        this.delta = (this.b * this.b) - (4 * this.a * this.c);
    }

    private void validarDelta() throws Exception
    {
        if(this.delta < 0) {
            throw new Exception("ERRO: Valores de A, B e C resultam num Delta negativo.\n");
        }
    }

    private double calcularX1()
    {
        return (
            ((this.b * -1) + Math.sqrt(this.delta))
            /
            (2 * this.a)
        );
    }

    private double calcularX2()
    {
        return (
            ((this.b * -1) - Math.sqrt(this.delta))
            /
            (2 * this.a)
        );
    }
}
